L1Trigger/DTTriggerPhase2/src/RPCIntegrator.cc

```diff
@@ -0,0 +1,260 @@
+  if (m_debug_)
+    std::cout << "Creating DT TP out of RPC recHits" << std::endl;
+  int rpc_wheel = rpcDetId.ring();             // In barrel, wheel is accessed via ring() method ([-2,+2])
+  int trigger_sector = rpcDetId.sector() - 1;  // DT Trigger sector:[0,11] while RPC sector:[1,12]
+  int rpc_station = rpcDetId.station();
+  int rpc_layer = rpcDetId.layer();
+  int rpc_trigger_phi = phiInDTTPFormat(rpc_global_phi, rpcDetId.sector());
+  int rpc_trigger_phiB = (phiB == -10000) ? phiB : (int)round(phiB * m_dt_phiB_granularity_);
+  int rpc_quality = -1;  // dummy for rpc
+  int rpc_index = 0;     // dummy for rpc
+  return L1Phase2MuDTPhDigi(rpc_bx,
+                            rpc_wheel,
+                            trigger_sector,
+                            rpc_station,
+                            rpc_layer,  //this would be the layer in the new dataformat
+                            rpc_trigger_phi,
+                            rpc_trigger_phiB,
+                            rpc_quality,
+                            rpc_index,
+                            rpc_time,
+                            -1,  // no chi2 for RPC
+                            rpc_flag);
+}
+
+double RPCIntegrator::phiBending(RPCMetaprimitive* rpc_hit_1, RPCMetaprimitive* rpc_hit_2) {
+  // Adaptation of https://github.com/dtp2-tpg-am/cmssw/blob/AM_106X_dev/L1Trigger/DTTriggerPhase2/src/MuonPathAssociator.cc#L189
+  DTChamberId DT_chamber(rpc_hit_1->rpc_id.ring(), rpc_hit_1->rpc_id.station(), rpc_hit_1->rpc_id.sector());
+  LocalPoint lp_rpc_hit_1_dtconv = dtGeo_->chamber(DT_chamber)->toLocal(rpc_hit_1->global_position);
+  LocalPoint lp_rpc_hit_2_dtconv = dtGeo_->chamber(DT_chamber)->toLocal(rpc_hit_2->global_position);
+  double slope = (lp_rpc_hit_1_dtconv.x() - lp_rpc_hit_2_dtconv.x()) / distance_between_two_rpc_layers_;
+  double average_x = (lp_rpc_hit_1_dtconv.x() + lp_rpc_hit_2_dtconv.x()) / 2;
+  GlobalPoint seg_middle_global =
+      dtGeo_->chamber(DT_chamber)->toGlobal(LocalPoint(average_x, 0., 0.));  // for station 1 and 2, z = 0
+  double seg_phi = phi_DT_MP_conv(seg_middle_global.phi(), rpc_hit_1->rpc_id.sector());
+  double psi = atan(slope);
+  double phiB = hasPosRF_rpc(rpc_hit_1->rpc_id.ring(), rpc_hit_1->rpc_id.sector()) ? psi - seg_phi : -psi - seg_phi;
+  return phiB;
+}
+
+int RPCIntegrator::phiInDTTPFormat(double rpc_global_phi, int rpcSector) {
+  double rpc_localDT_phi;
+  rpc_localDT_phi = phi_DT_MP_conv(rpc_global_phi, rpcSector) * m_dt_phi_granularity_;
+  return (int)round(rpc_localDT_phi);
+}
+
+double RPCIntegrator::phi_DT_MP_conv(double rpc_global_phi, int rpcSector) {
+  // Adaptation of https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TTwinMux/src/RPCtoDTTranslator.cc#L349
+  if (rpcSector == 1)
+    return rpc_global_phi;
+  else {
+    if (rpc_global_phi >= 0)
+      return rpc_global_phi - (rpcSector - 1) * Geom::pi() / 6.;
```

Comment:
magic numbers
use M_PI for pi
better to store 1/6. (multiplication is faster than division)